### DISCUSSION TITLE
Calculate function is not working anymore and result is not displayed

### DISCUSSION THREAD

**Discussion by donacrisante** [neuefische/questions#672] [Web] [answered]
Hi there!
I've been stuck on a problem all morning and it's that my CO2 calculation is no longer working. In my EntryForm component, the CO2 emissions of my journey are calculated depending on the means of transport and displayed when you click on "Calculate your impact". My goal was to have this result displayed on the JourneyList page when you click on "Add" and land on the JourneyList page. It kept telling me "result is undefined". So I changed the following: 
Put the state [result, setResult] in the App.js (it was in the EntryForm before), used useLocalStorageState instead of useState and passed "result" and "setResult" as props in the App.js, pages/index.js, EntryForm.js, EntryList.js and pages/journeyList/index.js. But it still doesn't work. The error message is "setResult is not a function". After trying a lot and having another student in my group look at it for a long time, I am posting my question here. Perhaps someone here can help me.

Here my PR:
https://github.com/donacrisante/Capstone-project/pull/10

Thank you!

**Comment by Kheirah** [comment, marked as answer]
In your `_app.js` when passing props down to the `Component` you accidentally passed `setResult` by calling it without arguments. Instead of this:
```js
setResult={() = setResult()}
```
you want this:
```js
setResult={setResult}
```
This is one way of solving the isse.
Another one would be to accept a parameter, but let's stay small and simple for now.

One more thing to change is this line:
```js
const [result, setResult] = useLocalStorageState("result", {
    defaultValue: 0, //this line changed from an array to a simple number value
    }); 
```
You don't need an array, I believe.

If you implement these changes, your result should appear properly on both pages.
But you will have to look into the console right after because there are some errors left to fix 🐱

**Comment by donacrisante** [comment]
Hi Ernst, thanks a lot for your answer. 
I originally passed the prop down as you just suggested, but I hadn't passed it down into the index.js. Then I passed it down in the index.js and changed it again in the App.js at the same time. That's why it never worked! 

I don't know what errors you're talking about, but when it comes to the toggle function, I've just managed that too. Now toggling also works ;)

**Reply by Kheirah** [reply]
Mhh, looks like the errors don't appear anymore. I saw something about svg properties. Nevermind then =)
